### PR TITLE
Allow underscores in names

### PR DESCRIPTION
**Pull request by stanimirovv**
NOTE: SInce the fix is getting bulky and the changes are mostly the same I would like some feedback fight now so that I have less changes to apply.

- [X] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type

- [X] Bugfix

## What is the current behavior?

Continuation of: https://github.com/nestjs/nest-cli/pull/1551
Underscores in names are not supported due to dasherize.

## What is the new behavior?
Underscores in names are supproted

## Does this PR introduce a breaking change?
- [X] No

### CODE REVIEW

**Comment by stanimirovv** [comment]
@micalevisk can you please review when you have the opportunity ? Not all files are fixed, but since the changes are somewhat repeatable I want to get a review so I have less things to redo :) 

Much appreciated !

File: src/lib/application/application.factory.test.ts [hunk 1]
```diff
@@ -91,15 +101,51 @@ describe('Application Factory', () => {
...
       ]);
 
-      expect(JSON.parse(tree.readContent('/awesome-project/package.json'))).toMatchObject({
+      expect(
+        JSON.parse(tree.readContent('/awesome-project/package.json')),
+      ).toMatchObject({
         name: 'awesome-project',
       });
     });
+    it('should keep backspaces', async () => {
```
Comment (micalevisk) [inline src/lib/application/application.factory.test.ts:110]: shouldn't be
```suggestion
    it('should keep underscores', async () => {
```

File: src/lib/decorator/decorator.factory.test.ts [hunk 2]
```diff
@@ -48,59 +52,85 @@ describe('Decorator Factory', () => {
...
-      files.find(filename => filename === '/bar-baz/foo/foo.decorator.ts'),
+      files.find((filename) => filename === '/bar-baz/foo/foo.decorator.ts'),
     ).not.toBeUndefined();
     expect(tree.readContent('/bar-baz/foo/foo.decorator.ts')).toEqual(
       "import { SetMetadata } from '@nestjs/common';\n" +
         '\n' +
         "export const Foo = (...args: string[]) => SetMetadata('foo', args);\n",
     );
   });
+  it('should keep underscores', async () => {
```
Comment (micalevisk) [inline src/lib/decorator/decorator.factory.test.ts:104]:
```suggestion
  it("should keep underscores on application's name", async () => {
```

File: src/lib/filter/filter.factory.test.ts [hunk 3]
```diff
@@ -86,14 +96,37 @@ describe('Filter Factory', () => {
     );
   });
 
-  it('should manage path to dasherize', async () => {
+  it('should keep backspaces', async () => {
```
Comment (micalevisk) [inline src/lib/filter/filter.factory.test.ts:99]:
```suggestion
  it("should keep underscores on file's name only", async () => {
```

File: src/lib/application/application.factory.test.ts (hunk 1 above)
Comment (stanimirovv) [reply src/lib/application/application.factory.test.ts:110]: fixed in all places.

File: src/lib/decorator/decorator.factory.test.ts [hunk 4]
```diff
@@ -48,59 +52,85 @@ describe('Decorator Factory', () => {
...
-      files.find(filename => filename === '/bar-baz/foo/foo.decorator.ts'),
+      files.find((filename) => filename === '/bar-baz/foo/foo.decorator.ts'),
     ).not.toBeUndefined();
     expect(tree.readContent('/bar-baz/foo/foo.decorator.ts')).toEqual(
       "import { SetMetadata } from '@nestjs/common';\n" +
         '\n' +
         "export const Foo = (...args: string[]) => SetMetadata('foo', args);\n",
     );
   });
+  it("should keep underscores on application's name", async () => {
```
Comment (micalevisk) [inline src/lib/decorator/decorator.factory.test.ts:104]:
```suggestion
  it("should keep underscores on decorator's path", async () => {
```

File: src/lib/filter/filter.factory.test.ts [hunk 5]
```diff
@@ -86,14 +96,37 @@ describe('Filter Factory', () => {
     );
   });
 
-  it('should manage path to dasherize', async () => {
+  it('should keep underscores on path only', async () => {
```
Comment (micalevisk) [inline src/lib/filter/filter.factory.test.ts:99]:
```suggestion
  it("should keep underscores on filter's path", async () => {
```

File: src/lib/gateway/gateway.factory.test.ts [hunk 6]
```diff
@@ -99,15 +107,41 @@ describe('Gateway Factory', () => {
         '}\n',
     );
   });
-  it('should manage path to dasherize', async () => {
+  it('should keep underscores', async () => {
```
Comment (micalevisk) [inline src/lib/gateway/gateway.factory.test.ts:110]:
```suggestion
  it("should keep underscores on gateway's path", async () => {
```

File: src/lib/guard/guard.factory.test.ts [hunk 7]
```diff
@@ -132,15 +142,43 @@ describe('Guard Factory', () => {
     );
   });
 
+  it('should keep underscore in path and name', async () => {
```
Comment (micalevisk) [inline src/lib/guard/guard.factory.test.ts:145]:
```suggestion
  it("should keep underscores guad's path", async () => {
```

I'm assuming that the 'name' is `BarGuard`

Comment (stanimirovv) [reply src/lib/guard/guard.factory.test.ts:145]: I've followed this for all tests and refactored the existing ones.

**Review by micalevisk — APPROVED** [review]
well done

image

image

(the above will be fixed by https://github.com/nestjs/nest-cli/pull/1551)

**Comment by kamilmysliwiec** [comment]
Once conflicts are resolved, it should be good to go!

**Comment by stanimirovv** [comment]
@kamilmysliwiec I've pushed a commit resolving the merge conflicts

**Comment by kamilmysliwiec** [comment]
LGTM! Great job!

**Pull request merged**